thriftbp/client_pool.go

```diff
@@ -169,6 +169,25 @@ type ClientPoolConfig struct {
 	BreakerConfig *breakerbp.Config
 }
 
+// Validate checks the ClientPoolConfig for any missing or erroneous values. It
+// returns an errorsbp.Batch to be combineable with other error batching.
+//
+// This method is designated to be used when passing a configuration to
+// NewBaseplateClientPool, for NewCustomClientPool other constraints apply.
```

Comment:
I wonder if it makes more sense to just have these validations split between `NewBaseplateClientPool` and `NewCustomClientPool` as appropriate rather than adding a function to the Config itself?

Reply:
I think so, because from an API point of view, having the method defined on the struct gives you the impression, it's always okay to call it, which it's not.

Reply:
What about having a 
```
ValidateBaseplateClientPoolConfig
```
method? :thinking: 

Alternatively
```
type BaseplateClientPoolConfig {
  ClientPoolConfig
}

func (c *BaseplateClientPoolConfig) Validate() error {
  // ...
}
```

Reply:
`ValidateBaseplateClientPoolConfig` seems reasonable.  I actually _kind of_ like the second option but it might be a little too cute 😅 

@fishy do you have any opinions?

Reply:
I can see the value of adding a function to the config struct itself, so that users can do the validation without actually trying to use it (e.g. they can write static checking tools, kinda). With that spirit it might be better to split it into two functions, `ValidateCustom` which doesn't check `Addr`, and then `Validate` which calls `ValidateCustom` and also checks `Addr`.

Reply:
If we go the embedding route, it might be easier to just do typedef instead? e.g.

```
type BaseplateClientPoolConfig Config
```

Then we can override its `Validate` implementation.

Reply:
but that gets tricky as in the original `ClientPoolConfig` some of the fields are actually never used (they are only used by `BaseplateClientPoolConfig`). so maybe just split into two functions?